### PR TITLE
Feature/set each attribute

### LINKED ISSUE #24: Set every Attribute By Rolling Dice
Following the same logic as in #20 
Now we have to implement all endpoints for all 6 attributes (5 left): Dexterity (DEX), Constitution (CON), Intelligence (INT), Wisdom (WIS), Charisma (CHA)
Those are the saving throw and skills that are related to those attributes

There are 6 types of saving throws and each is equivalent to the associated attribute (the skill saving throw is equal to the DEX modifier and so on).

|      Skills           | **DEX** | **WIS** | **INT** | **CHA** | **STR** |
|-----------------|---------|---------|---------|---------|---------|
| Acrobatics      |    X    |         |         |         |         |
| Animal Handling |         |    X    |         |         |         |
| Arcana          |         |         |    X    |         |         |
| Athletics       |         |         |         |         |    X    |
| Deception       |         |         |         |    X    |         |
| History         |         |         |    X    |         |         |
| Insight         |         |    X    |         |         |         |
| Intimidation    |         |         |         |    X    |         |
| Investigation   |         |         |    X    |         |         |
| Medicine        |         |    X    |         |         |         |
| Nature          |         |         |    X    |         |         |
| Perception      |         |    X    |         |         |         |
| Perfomance      |         |         |         |    X    |         |
| Persuasion      |         |         |         |    X    |         |
| Religion        |         |         |    X    |         |         |
| Sleight of Hand |    X    |         |         |         |         |
| Stealth         |    X    |         |         |         |         |
| Survival        |         |    X    |         |         |         |

### PR DESCRIPTION
**Pull request by zetTtai**
Closes #24

### CODE REVIEW

File: DnDCharacterSheet/Models/Attribute.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using Enums;
+
+namespace Models
+{
```
Comment (KiyanDev) [inline DnDCharacterSheet/Models/Attribute.cs:4]: use file scoped namespaces, this is an unnecessary level of identation, you can configure this in visual studio settings

File: DnDCharacterSheet/Models/Sheet.cs [hunk 2]
```diff
@@ -14,27 +14,47 @@ public Sheet()
...
+                .Select( attribute => new Attribute
+                {
+                    Name = attribute,
+                    Value = "",
+                    Modifier = "",
+                }).ToList();
+
             Dictionary<string, CharacterAttributes> staticSkills = new()
             {
                 {"Athletics", CharacterAttributes.STR },
```
Comment (KiyanDev) [inline DnDCharacterSheet/Models/Sheet.cs:43]: Important: All of these hardcoded values should be either an enum or a class

File: DnDCharacterSheet/Services/SheetService.cs [hunk 3]
```diff
@@ -8,15 +9,15 @@ public class SheetService : ISheetService
     {
         private IAttributeSettingStrategy? _attributeSettingStrategy;
 
-        public void SetStrategy(IAttributeSettingStrategy strategy)
+        public void SetAttributeSettingStrategy(IAttributeSettingStrategy strategy)
         {
             _attributeSettingStrategy = strategy;
```
Comment (KiyanDev) [inline DnDCharacterSheet/Services/SheetService.cs:14]: check for null

File: DnDCharacterSheet/Services/SheetService.cs [hunk 4]
```diff
@@ -8,15 +9,15 @@ public class SheetService : ISheetService
...
         }
 
-        public Sheet SetStrengthAttribute(Sheet sheet, int value)
+        public Sheet SetAttribute(Sheet sheet, int value)
         {
             return _attributeSettingStrategy != null ?
-                _attributeSettingStrategy.SetStrengthAttribute(sheet, value)
+                _attributeSettingStrategy.SetAttribute(sheet, value)
                 : throw new Exception(Constants.SheetService.NoStrategyError);
         }
```
Comment (KiyanDev) [inline DnDCharacterSheet/Services/SheetService.cs:22]: all of these code should not be allowed to work with a null reference, therefore avoid processing any nullable code

File: DnDTests/Converters/SheetConverterTest.cs [hunk 5]
```diff
@@ -33,7 +38,15 @@ public void Convert_Sheet_ReturnsSheetDTO()
             SheetDTO expected = new()
             {
                 Id = 1,
-                StrengthAttribute = "",
+                Attributes =
+                [
+                    new AttributeDTO { Modifier = "", Value = "", Name = "STR" },
```
Comment (KiyanDev) [inline DnDTests/Converters/SheetConverterTest.cs:43]: do not hardcode "" , instead use string.Empty

File: DnDTests/Converters/SheetConverterTest.cs [hunk 6]
```diff
@@ -58,13 +70,13 @@ public void Convert_List_ReturnsSheetDTOList()
...
-                new() { Id = 2, StrengthAttribute = "" },
-                new() { Id = 3, StrengthAttribute = "" },
+                new() { Id = 1},
+                new() { Id = 2},
+                new() { Id = 3},
             ];
 
             // Act
-            List<SheetDTO> actual = (List<SheetDTO>)_converter.Convert(sheets);
+            List<SheetDTO> actual = _converter.Convert(sheets).ToList();
```
Comment (KiyanDev) [inline DnDTests/Converters/SheetConverterTest.cs:79]: use var pattern

File: DnDTests/Services/SheetServiceTest.cs [hunk 7]
```diff
@@ -19,35 +19,41 @@ public void OneTimeSetUp()
...
+        public void SetAttribute_StrategySet_ReturnsSheet()
         {
             // Arrange
-            _service.SetStrategy(_attributeSettingStrategyMock.Object);
-            Sheet expected = new();
+            _service.SetAttributeSettingStrategy(_attributeSettingStrategyMock.Object);
 
             // Act
-            Sheet actual = _service.SetStrengthAttribute(new Sheet(), 3);
+            try
```
Comment (KiyanDev) [inline DnDTests/Services/SheetServiceTest.cs:34]: take care in formatting please, use cleanup in the IDE either before pushing manually or run the code cleanup directly with a githook before pushing :)

File: DnDTests/Services/UtilsServiceTest.cs [hunk 8]
```diff
@@ -1,13 +1,18 @@
...
 using Services;
+using Strategies;
 
 namespace DnDTests.Services
 {
     internal class UtilsServiceTest
     {
         private IUtilsService _service;
+        private readonly string _invalidAttributeError = "Attribute must be STR, DEX, CON, INT, WIS or CHA";
+
```
Comment (KiyanDev) [inline DnDTests/Services/UtilsServiceTest.cs:15]: i've already read this somewhere ( i know where ;) ) do you really need this here?

File: DnDTests/Services/UtilsServiceTest.cs [hunk 9]
```diff
@@ -84,55 +89,139 @@ public void ValueToAttributeModifier_EqualToTen_ReturnsString()
...
         }
 
-        [Test]
-        public void ModifyCapabilities_ReturnsCapabilityList()
+        [TestCase(CharacterAttributes.STR)]
+        [TestCase(CharacterAttributes.DEX)]
+        [TestCase(CharacterAttributes.CON)]
+        [TestCase(CharacterAttributes.INT)]
+        [TestCase(CharacterAttributes.WIS)]
+        [TestCase(CharacterAttributes.CHA)]
```
Comment (KiyanDev) [inline DnDTests/Services/UtilsServiceTest.cs:97]: you can do this iterating over the character attributes witth a testSource so this way you would not need to update this test if you include more characters attributes

File: DnDTests/Services/UtilsServiceTest.cs [hunk 10]
```diff
@@ -84,55 +89,139 @@ public void ValueToAttributeModifier_EqualToTen_ReturnsString()
...
 
-            List<Capability> expected = [
-                new()
-                {
-                    Name = "Test1",
-                    AssociatedAttribute = CharacterAttributes.STR,
-                    Value = "+2",
-                },
-                new()
+            // Act
```
Comment (KiyanDev) [inline DnDTests/Services/UtilsServiceTest.cs:121]: These are redundant and provide no information, unless the test has really such a complex case that these are needed ( and in which case is an indication that refactoring is needed )

Comment (zetTtai) [reply DnDTests/Services/UtilsServiceTest.cs]: This have been modified in later commits, I think that new ModifyCapabilities_ReturnsCapabilityList test is no longer redundant

File: DnDTests/Services/UtilsServiceTest.cs (hunk 8 above)
Comment (zetTtai) [reply DnDTests/Services/UtilsServiceTest.cs]: This text has been removed, but was used to check if the expected error message is equal to the one set in Constants.cs.

**Pull request closed without merging**